gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/jsr223/JavaTranslator.java

```diff
@@ -266,6 +267,23 @@ private Object invokeMethod(final Object delegate, final Class<?> returnType, fi
                                 }
                             }
                         }
+
+                        // special case has() where the first arg is null - sometimes this can end up with the T being
+                        // null and in 3.5.x that generates an exception which raises badly in the translator. it is
+                        // safer to force this to the String form by letting this "found" version pass. In java this
+                        // form of GraphTraversal can't be produced because of validations for has(T, ...) but in
+                        // other language it might be allowed which means that has((T) null, ...) from something like
+                        // python will end up as has((String) null, ...) which filters rather than generates an
+                        // exception. calling the T version even in a non-JVM language seems odd and more likely the
+                        // caller was shooting for the String one, but ugh who knows. this issue showcases again how
+                        // badly bytecode should either change to use gremlin-language and go away or bytecode should
+                        // get a safer way to be translated to a traversal with more explicit calls that don't rely
+                        // on reflection.
+                        if (methodName.equals(GraphTraversal.Symbols.has) && newArguments.length > 0 && null == newArguments[0] &&
+                            method.getParameterTypes()[0].isAssignableFrom(org.apache.tinkerpop.gremlin.structure.T.class)) {
```

Comment:
Please correct me if I am wrong here but the tests in this PR do not test for this condition that is mentioned in this comment `has((T) null, ...)` . Could we please add a test for it?